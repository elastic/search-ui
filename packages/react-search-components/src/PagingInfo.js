import PropTypes from "prop-types";
import React from "react";

function PagingInfo({ end, searchTerm, start, totalResults }) {
  return (
    <div className="sui-paging-info">
      Showing{" "}
      <strong>
        {start} - {end}
      </strong>{" "}
      out of <strong>{totalResults}</strong> for: <em>{searchTerm}</em>
    </div>
  );
}

PagingInfo.propTypes = {
  end: PropTypes.number.isRequired,
  searchTerm: PropTypes.string.isRequired,
  start: PropTypes.number.isRequired,
  totalResults: PropTypes.number.isRequired
};

export default PagingInfo;
