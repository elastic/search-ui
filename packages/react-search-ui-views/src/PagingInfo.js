import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function PagingInfo({
  className,
  end,
  searchTerm,
  start,
  totalResults,
  ...rest
}) {
  return (
    <div className={appendClassName("sui-paging-info", className)} {...rest}>
      Showing{" "}
      <strong>
        {start} - {end}
      </strong>{" "}
      out of <strong>{totalResults}</strong>
      {searchTerm && (
        <>
          {" "}
          for: <em>{searchTerm}</em>
        </>
      )}
    </div>
  );
}

PagingInfo.propTypes = {
  end: PropTypes.number.isRequired,
  searchTerm: PropTypes.string.isRequired,
  start: PropTypes.number.isRequired,
  totalResults: PropTypes.number.isRequired,
  className: PropTypes.string
};

export default PagingInfo;
