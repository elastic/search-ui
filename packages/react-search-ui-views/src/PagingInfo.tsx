import type { SearchContextState } from "@elastic/search-ui";
import React from "react";
import { Rename, BaseContainerProps } from "./types";

import { appendClassName } from "./view-helpers";

export type PagingInfoContainerContext = Pick<
  SearchContextState,
  "pagingStart" | "pagingEnd" | "resultSearchTerm" | "totalResults"
>;

export type PagingInfoViewProps = Rename<
  BaseContainerProps & PagingInfoContainerContext,
  {
    pagingStart: "start";
    resultSearchTerm: "searchTerm";
    pagingEnd: "end";
  }
>;

export type PagingInfoContainerProps = BaseContainerProps &
  PagingInfoContainerContext & {
    view?: React.ComponentType<PagingInfoViewProps>;
  };

function PagingInfo({
  className,
  end,
  searchTerm,
  start,
  totalResults,
  ...rest
}: PagingInfoViewProps & React.HTMLAttributes<HTMLDivElement>) {
  return (
    <div className={appendClassName("sui-paging-info", className)} {...rest}>
      Showing{" "}
      <strong>
        {start} - {end}
      </strong>{" "}
      out of <strong>{totalResults}</strong>
      {searchTerm && (
        <>
          {" "}
          for: <em>{searchTerm}</em>
        </>
      )}
    </div>
  );
}

export default PagingInfo;
