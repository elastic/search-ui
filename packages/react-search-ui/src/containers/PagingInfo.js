import PropTypes from "prop-types";
import { Component } from "react";
import { withSearch } from "..";
import { PagingInfo } from "@elastic/react-search-ui-views";

export class PagingInfoContainer extends Component {
  static propTypes = {
    // Props
    className: PropTypes.string,
    view: PropTypes.func,
    // State
    pagingStart: PropTypes.number.isRequired,
    pagingEnd: PropTypes.number.isRequired,
    resultSearchTerm: PropTypes.string.isRequired,
    totalResults: PropTypes.number.isRequired
  };

  render() {
    const {
      className,
      pagingStart,
      pagingEnd,
      resultSearchTerm,
      totalResults,
      view,
      ...rest
    } = this.props;

    const View = view || PagingInfo;

    return View({
      className,
      searchTerm: resultSearchTerm,
      start: pagingStart,
      end: pagingEnd,
      totalResults: totalResults,
      ...rest
    });
  }
}

export default withSearch(
  ({ pagingStart, pagingEnd, resultSearchTerm, totalResults }) => ({
    pagingStart,
    pagingEnd,
    resultSearchTerm,
    totalResults
  })
)(PagingInfoContainer);
