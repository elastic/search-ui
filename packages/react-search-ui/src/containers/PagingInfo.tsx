import React, { Component } from "react";
import { withSearch } from "..";
import {
  PagingInfoContainerProps,
  PagingInfo,
  PagingInfoViewProps,
  PagingInfoContainerContext
} from "@elastic/react-search-ui-views";

export class PagingInfoContainer extends Component<PagingInfoContainerProps> {
  render() {
    const {
      className,
      pagingStart,
      pagingEnd,
      resultSearchTerm,
      totalResults,
      view,
      ...rest
    } = this.props;

    const View = view || PagingInfo;
    const viewProps: PagingInfoViewProps = {
      className,
      searchTerm: resultSearchTerm,
      start: pagingStart,
      end: pagingEnd,
      totalResults: totalResults,
      ...rest
    };

    return <View {...viewProps} />;
  }
}

export default withSearch<PagingInfoContainerProps, PagingInfoContainerContext>(
  ({ pagingStart, pagingEnd, resultSearchTerm, totalResults }) => ({
    pagingStart,
    pagingEnd,
    resultSearchTerm,
    totalResults
  })
)(PagingInfoContainer);
