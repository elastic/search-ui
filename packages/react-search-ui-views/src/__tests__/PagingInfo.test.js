import React from "react";
import PagingInfo from "../PagingInfo";
import { shallow } from "enzyme";

const props = {
  end: 20,
  searchTerm: "grok",
  start: 0,
  totalResults: 1000
};

it("renders correctly", () => {
  const wrapper = shallow(<PagingInfo {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const wrapper = shallow(
    <PagingInfo className={customClassName} {...props} />
  );
  const { className } = wrapper.childAt(0).props();
  expect(className).toEqual("sui-paging-info test-class");
});

it("does not render a higher end than the total # of results", () => {
  const wrapper = shallow(<PagingInfo {...props} totalResults={15} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders ScreenReaderStatus with the correct message", () => {
  const announceToScreenReader = jest.fn();
  const wrapper = shallow(<PagingInfo {...props} start={41} end={80} />);

  wrapper.find("ScreenReaderStatus").prop("render")(announceToScreenReader);
  expect(announceToScreenReader).toHaveBeenCalledWith(
    'Showing 41 to 80 results out of 1000, searching for "grok".'
  );

  // Should not call out search term if one isn't present
  wrapper.setProps({ searchTerm: "" });
  wrapper.find("ScreenReaderStatus").prop("render")(announceToScreenReader);
  expect(announceToScreenReader).toHaveBeenCalledWith(
    "Showing 41 to 80 results out of 1000"
  );
});
