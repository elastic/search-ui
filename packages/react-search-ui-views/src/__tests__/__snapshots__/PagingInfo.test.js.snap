// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sui-paging-info"
>
  Showing
   
  <strong>
    0
     - 
    20
  </strong>
   
  out of 
  <strong>
    1000
  </strong>
   
  for: 
  <em>
    grok
  </em>
</div>
`;
